src/rfq/protocols/bebop/client.rs

```diff
@@ -906,4 +927,107 @@ mod tests {
         assert_eq!(res.base_token, params.token_in);
         assert_eq!(res.quote_token, params.token_out);
     }
+
+    #[tokio::test]
+    async fn test_bebop_quote_timeout() {
```

Comment:
Using this test I think I reproduced the issue? I would get `error sending request for url` if I don't set a timeout like I did above.

Adding the timeout, I get two behaviours:
1. If no response before the timeout, I get timeout error.
2. If it just takes a bit of time, but still less than the timeout, then I get a different (but expected) parsing error...

I tried to get rid of the parsing error which shows up because I made a nonsense response here `let response = r#"response"#;` but I couldn't make it work and wasn't sure how much time I should lose on this...